diff --git a/win32/updateAndBuildAll.py b/win32/updateAndBuildAll.py
index db1da367c..d0cf77819 100644
--- a/win32/updateAndBuildAll.py
+++ b/win32/updateAndBuildAll.py
@@ -21,6 +21,9 @@
 
 MSBUILD_LOCATIONS = [r'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin',
                      r'C:/Program Files (x86)/Microsoft Visual Studio/2022/Enterprise/MSBuild/Current/Bin',
+                     r'C:/Program Files/Microsoft Visual Studio/2022/Community/MSBuild/Current/Bin',
+                     r'C:/Program Files/Microsoft Visual Studio/2022/Enterprise/MSBuild/Current/Bin',
+                     r'/cygdrive/c/Program Files/MSBuild/14.0/Bin/',
                      r'/cygdrive/c/Program Files (x86)/MSBuild/14.0/Bin/']
 
 MSBUILD = 'MSBuild.exe'
@@ -81,20 +84,20 @@ def buildTSKAll():
         print("Please set the TSK_HOME environment variable")
         sys.exit(1)
 
-    if not MINIMAL:
-        if(passed):
-            buildTSK(32, "Release")
-        if(passed):
-            buildTSK(64, "Release_NoLibs")
+    # if not MINIMAL:
+    #     if(passed):
+    #         buildTSK(32, "Release")
+    #     if(passed):
+    #         buildTSK(64, "Release_NoLibs")
 
 
     # MINIMAL is 64-bit for Autopsy and 32-bit with no deps for logical imager et al.
-    if(passed):
-        buildTSK(32, "Release_NoLibs")
-    if(passed):
-        BuildXPNoLibsFilePath = os.path.join(TSK_HOME, "build_xpnolibs")
-        if os.path.exists(BuildXPNoLibsFilePath):
-            buildTSK(32, "Release_XPNoLibs")
+    # if(passed):
+    #     buildTSK(32, "Release_NoLibs")
+    # if(passed):
+    #     BuildXPNoLibsFilePath = os.path.join(TSK_HOME, "build_xpnolibs")
+    #     if os.path.exists(BuildXPNoLibsFilePath):
+    #         buildTSK(32, "Release_XPNoLibs")
     if(passed):
         buildTSK(64, "Release")
 
