diff --git a/win32/updateAndBuildAll.py b/win32/updateAndBuildAll.py
index db1da367c..26920e569 100644
--- a/win32/updateAndBuildAll.py
+++ b/win32/updateAndBuildAll.py
@@ -20,8 +20,11 @@
 import traceback
 
 MSBUILD_LOCATIONS = [r'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin',
+                     r'C:/Program Files/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin',
                      r'C:/Program Files (x86)/Microsoft Visual Studio/2022/Enterprise/MSBuild/Current/Bin',
-                     r'/cygdrive/c/Program Files (x86)/MSBuild/14.0/Bin/']
+                     r'C:/Program Files/Microsoft Visual Studio/2022/Enterprise/MSBuild/Current/Bin',
+                     r'/cygdrive/c/Program Files (x86)/MSBuild/14.0/Bin/',
+                     r'/cygdrive/c/Program Files/MSBuild/14.0/Bin/']
 
 MSBUILD = 'MSBuild.exe'
 
